org.jacoco.core/src/org/jacoco/core/data/ExecutionDataWriter.java

```diff
@@ -28,7 +28,7 @@
 
 	static {
 		// Runtime initialize to ensure javac does not inline the value.
-		FORMAT_VERSION = 0x1007;
+		FORMAT_VERSION = 0x1008;
```

Comment:
I'm wondering whether version increment is really required, if the only change - is an end of try-catch blocks and not number and placement of probes? Otherwise I'm afraid of consequences of this incompatible change for downstream projects as already mentioned in https://github.com/jacoco/jacoco/issues/626#issuecomment-349073113 - when we did increment last time several years ago, it was quite disruptive.

Also in backlog we have some other changes that require increment, but were considered minor to justify it. So if underlying issue here justifies increment, then we'd better do them all together.

Reply:
To be honest, I'm not quite sure whether the increment is necessary. I remember @marchof said it was needed for [#265](https://github.com/jacoco/jacoco/issues/265#issuecomment-68149109) but you are correct that the number of probes and their placement doesn't really change.

Reply:
Actually not obvious that number and placement do not change - I was not and is not sure about this and that was a reason to ask.

Also I have feeling that our unit tests unfortunately are not exhaustive in regards to number and placement of probes and especially in case of complex CFGs involving try-catch blocks, and unfortunately we don't have assertions about number of probes in validation tests and they are probably not exhaustive too.

And since you're also unsure, we'll need to study this more carefully, e.g. by counting number of probes on a big set of class files before and after this change.

Reply:
@allenhair in fact after merge of #266 @marchof reverted increment of version - 82f40d7151ef0f99e18573a4e6b248eba6abc3ad :wink:

And here I come to following conclusions: move of start and end labels of try-catch blocks between end and beginning of probe doesn't affect original labels and so doesn't affect MethodInstrumenter, i.e. placement and number of probes, as well as MethodAnalyzer, i.e. result of analysis, and affects only final class after instrumentation. I guess this is exactly what is described as "leads to semantically identical probes" in 82f40d7151ef0f99e18573a4e6b248eba6abc3ad.

@marchof Am I right? WDYT?

Reply:
@Godin @allenhair If the change in #266 did not change the number of probes this new handling for the end labels will also not change the number of probes. So no need to change exec version.
Also if we would do so we should combine this with other changes in our backlog (see @Godin 's comment).